src/prototype_creep_resources.js

```diff
@@ -13,10 +13,12 @@ Creep.prototype.universalBeforeStorage = function() {
   if (this.room.controller && (this.room.controller.ticksToDowngrade < CONTROLLER_DOWNGRADE[this.room.controller.level] / 10 || this.room.controller.level === 1)) {
     methods.push(Creep.upgradeControllerTask);
   }
-  const universals = this.room.findPropertyFilter(FIND_MY_CREEPS, 'memory.role', ['universal']).map((creep) => creep.name);
-  universals.sort();
-  if (universals.indexOf(this.name) < 2) {
-    methods.push(Creep.transferEnergy);
+  if (!this.room.isConstructingSpawn()) {
```

Comment:
why not transferEnergy if room.isConstructingSpawn?

Reply:
> maybe if some config is set like

OK, can do.

Reply:
> why not transferEnergy if room.isConstructingSpawn?

because transferEnergy have higher priority than constructTask.

means, even if you are rebuilding spawn, the universals will still fill your precious energy into extensions, which can be a disaster.